### PR TITLE
Add lis2mdl sensor support in SensorTile.box

### PR DESCRIPTION
**Pull request by avisconti**
Add lis2mdl sensor support in SensorTile.box.

This PR depends to #19430

### CODE REVIEW

**Review by erwango — APPROVED** [review]

File: drivers/sensor/lis2mdl/Kconfig [hunk 1]
```diff
@@ -72,4 +72,8 @@ config LIS2MDL_MAG_ODR_100
 
 endchoice
 
+config LIS2MDL_SPI_FULL_DUPLEX
+	bool "Enable SPI 4wire mode (separated MISO and MOSI lines"
```
Comment (MaureenHelm) [inline drivers/sensor/lis2mdl/Kconfig]: Missing ")"

File: drivers/sensor/lis2mdl/lis2mdl_spi.c [hunk 2]
```diff
@@ -122,6 +122,13 @@ int lis2mdl_spi_init(struct device *dev)
 		cfg->gpio_cs_port, cfg->cs_gpio);
 #endif
 
+#if CONFIG_LIS2MDL_SPI_FULL_DUPLEX
+	/* Set SPI 4wires */
+	if (lis2mdl_spi_mode_set(data->ctx, LIS2MDL_SPI_4_WIRE) < 0) {
```
Comment (MaureenHelm) [inline drivers/sensor/lis2mdl/lis2mdl_spi.c]: why do this twice (once in `lis2mdl_init` and again here)?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/lis2mdl/Kconfig (hunk 1 above)
Comment (avisconti) [reply drivers/sensor/lis2mdl/Kconfig]: ack
Sorry for the typo

Comment (avisconti) [reply drivers/sensor/lis2mdl/Kconfig]: changed and repushed!

File: drivers/sensor/lis2mdl/lis2mdl_spi.c (hunk 2 above)
Comment (avisconti) [reply drivers/sensor/lis2mdl/lis2mdl_spi.c]: @MaureenHelm 
Are you ok with this explanation?

**Comment by avisconti** [comment]
@MaureenHelm 
I would like to get this PR merged for v2.1.0.
Can you take a look if all ur requests have been properly addressed?

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**